src/acftbias/GsiAircraftBiasConverter.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ *  * (C) Copyright 2023 NOAA/NWS/NCEP/EMC
+ *   *
+ *    * This software is licensed under the terms of the Apache Licence Version 2.0
+ *     * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *      */
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <iostream>
+
+#include <Eigen/Dense>
+
+#include "eckit/config/LocalConfiguration.h"
+#include "eckit/config/YAMLConfiguration.h"
+#include "eckit/exception/Exceptions.h"
+#include "eckit/filesystem/PathName.h"
+
+#include "ioda/ObsGroup.h"
+#include "ioda/Engines/HH.h"
+
+#include "oops/util/missingValues.h"
+
+#include "GsiAircraftBiasReader.h"
+
+ioda::ObsGroup makeObsBiasObject(ioda::Group &empty_base_object,
+                                 const std::string & coeffile,
+                                 const std::vector<std::string> & tailIds,
+                                 const std::vector<std::string> & predictors) {
+  /// Predictors
+  int numPreds = predictors.size();
+  int numIds = tailIds.size();
+
+  /// Creating dimensions: n
+  ioda::NewDimensionScales_t newDims {
+      ioda::NewDimensionScale<int>("nvars", 1),
+      ioda::NewDimensionScale<int>("nrecs", numIds)
+  };
+
+  /// Construct an ObsGroup object, with 2 dimensions nrecs, nvars
+  ioda::ObsGroup ogrp = ioda::ObsGroup::generate(empty_base_object, newDims);
+
+  /// Create tail IDs and records variable
+  ioda::Variable tailIdsVar = ogrp.vars.createWithScales<std::string>("records",
+                                    {ogrp.vars["nrecs"]});
+  tailIdsVar.write(tailIds);
+
+  /// Create 2D bias coefficient variable
+  Eigen::ArrayXXf biascoeffs(numIds, numPreds*3);
+
+  readObsBiasCoefficients(coeffile, biascoeffs);
+
+  for (int i = 0; i < numPreds; ++i) {
+    /// Set up the creation parameters for the bias coefficients variable
+    ioda::VariableCreationParameters float_params;
+    float_params.chunk = true;               // allow chunking
+    float_params.compressWithGZIP();         // compress using gzip
+    float missing_value = util::missingValue(missing_value);
+    float_params.setFillValue<float>(missing_value);
+
+    // Access predictor coefficient values column and create variable
+    Eigen::ArrayXXf subVar = biascoeffs.col(i);
+
+    ioda::Variable biasVar = ogrp.vars.createWithScales<float>("biasCoefficients/"+predictors[i],
+                       {ogrp.vars["nvars"], ogrp.vars["nrecs"]}, float_params);
+    biasVar.writeWithEigenRegular(subVar);
+
+    // Access predictor observation error values column and create variable
+    Eigen::ArrayXXf subVarObError = biascoeffs.col(i+3);
+
+    ioda::Variable biasVarObError = ogrp.vars.createWithScales<float>(
+                                                "biasCoeffObError/"+predictors[i],
```

Comment:
I think we might just want `ObError` here but want others to chime in here @ADCollard @shlyaeva @BenjaminRuston @gmao-yzhu @huishao-r @gmao-msienkie